python/cudf/udf_cpp/shim.cu

```diff
@@ -720,3 +722,49 @@ make_definition_corr(BlockCorr, int64, int64_t);
 
 #undef make_definition_corr
 }
+
+/*
+NRT CUDA functions
+*/
+
+// Only used to allocate the right amount of space, see below
```

Comment:
This does not seem to be the case. "Below" I see a `meminfo_and_str` being allocated, then the locations of its members determined, and passed to `NRT_MemInfo_init`. That seems like more than just providing the required size of the allocation.